[seanfirefox] iP

seanfirefox opened the pull request:
## Duke Chatbot

has the following **features**:
- *Stores* tasks
* *Deletes* tasks
+ *Prompts* user on invalid inputs

with levels completed
1. ~~Level-1~~
2. ~~Level-2~~
3. ~~Level-3~~
4. ~~Level-4~~
5. ~~Level-5~~
6. ~~Level-6~~
7. ~~Level-7~~
8. ~~Level-8~~
9. ~~Level-9~~

Please use ``` java -jar duke.java ``` to test the jar file.

Tasks not done this week are:
- [x] A-Checkstyle
- [x] Level-10
- [ ] A-Varargs
:sweat_smile: 

>For those who cannot find my github
[My Github link](https://github.com/seanfirefox)

If you Java programmer, you can use it to practice Java too.
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

seanfirefox committed:
Level 1 Greet Echo Exit done

seanfirefox committed:
Level 2 Add List done

seanfirefox committed:
Level 3 Mark as Done done

seanfirefox committed:
Level 4 ToDos, Events, Deadlines done

seanfirefox committed:
A-TextUiTesting Automated Text UI Testing done

seanfirefox committed:
Change from if else to switch case

seanfirefox committed:
Level 5 Handle Errors done

seanfirefox committed:
Level 6 Delete done

seanfirefox committed:
Create directory and task file

seanfirefox committed:
Level 7 Save done

seanfirefox committed:
Level 8 Dates and Times done

seanfirefox committed:
Merge branch 'branch-Level-8'

seanfirefox committed:
Add TaskList usage to Duke

seanfirefox committed:
Add UI to usage

seanfirefox committed:
Add Command class

Jiayan-Lim commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,79 @@
+                switch (arr[0]) {
+                    case "list":
+                        taskList.listTask(arr);
+                        break;
+                    case "mark":
+                        taskList.markTask(true, arr, ui);
+                        storage.writeData(taskList);
+                        break;
+                    case "unmark":
+                        taskList.markTask(false, arr, ui);
+                        storage.writeData(taskList);
+                        break;
+                    case "todo":
+                        ToDo toDo = ToDo.generate(arr);
+                        taskList.addTask(toDo);
+                        storage.writeData(taskList);
+                        ui.printAddTaskMsg(taskList, toDo);
+                        break;
+                    case "deadline":
+                        DeadLine deadLine = DeadLine.generate(arr);
+                        taskList.addTask(deadLine);
+                        storage.writeData(taskList);
+                        ui.printAddTaskMsg(taskList, deadLine);
+                        break;
+                    case "event":
+                        Event event = Event.generate(arr);
+                        taskList.addTask(event);
+                        storage.writeData(taskList);
+                        ui.printAddTaskMsg(taskList, event);
+                        break;
+                    case "delete":
+                        if (arr.length < 2) {
+                            throw new DukeException("Invalid format, please give numbers");
+                        }
+                        taskList.deleteTask(arr[1], ui);
+                        storage.writeData(taskList);
+
+                        break;
+                    default:
+                        throw new DukeException("Unknown format");
```
This chunk of code has violated Java coding standards.  There should be no indentation for case and default clause.

seanfirefox replied:
Resolved it in later commits, thanks for pointing out!

Jiayan-Lim commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,61 @@
+
+    @Override
+    public String storeTaskString() {
+        return this.getTaskType() + " | " + this.getCheckedString() + " | " + this.getStr() + " | " + this.getStart() + " | " + this.getEnd();
```
Perhaps can introduce line wrapping to keep the line length shorter since the hard limit of line length is 120 chars.

seanfirefox replied:
Fixed

Jiayan-Lim commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,71 @@
+                String nextLine = scanner.nextLine();
+                String[] taskLine = nextLine.trim().split(" \\| ");
+                switch (taskLine[0]) {
+                    case "T":
+                        arrayList.add(ToDo.generateTask(taskLine));
+                        break;
+                    case "E":
+                        arrayList.add(Event.generateTask(taskLine));
+                        break;
+                    case "D":
+                        arrayList.add(DeadLine.generateTask(taskLine));
+                        break;
```
For switch-case statement, there should no indentation for case clause.

seanfirefox replied:
Fixed

Jiayan-Lim commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,71 @@
+    public void writeData(TaskList taskList) {
+        try {
+            FileWriter fileWriter = new FileWriter(filePath);
+            ArrayList<Task> arrayList = taskList.getStore();
```
Maybe you can use plural form when representing a collection of objects as mentioned in the java coding guidelines.

seanfirefox replied:
Fixed

Jiayan-Lim commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,51 @@
+
+    public void printDeleteTaskMsg(Task task, int size) {
+        System.out.println("Noted. I've removed this task:\n  " +
+                task.toString() +
+                "\nNow you have " + size + " tasks in the list.");
```
```suggestion
                task.toString() 
                + "\nNow you have " + size + " tasks in the list.");
```
Maybe you can break before an operator to improve readability as mentioned in the coding guidelines.

seanfirefox replied:
Fixed

Jiayan-Lim commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,4 @@
+abstract public class Command {
+    abstract public void execute(TaskList taskList, Ui ui, Storage storage);
```
Perhaps you can add some header comments for public class and its method.

seanfirefox replied:
Added javadocs in later commits.

Jiayan-Lim reviewed (commented):
LGTM! Most of the coding standards were adhered to! Just some nits to fix 😄

seanfirefox committed:
A-MoreOOP Use More OOP done

seanfirefox committed:
A-Packages Organise Into Packages done

seanfirefox committed:
Update paths to shell script

seanfirefox committed:
Merge branch 'add-gradle-support'

seanfirefox committed:
A-JUnit Add JUnit Tests done

seanfirefox committed:
Edit gradle classpath

seanfirefox committed:
A-JavaDoc Add JavaDoc comments done.

seanfirefox committed:
A-CodingStandard Following the Coding Standard done.

seanfirefox committed:
Format Javadoc comments

seanfirefox committed:
Level-9 Find done

seanfirefox committed:
Resolve merge conflict branch-A-CodingStandard

seanfirefox committed:
Resolve merge conflict branch-Level-9

seanfirefox committed:
Fix merge conflict errors

seanfirefox committed:
A-Checkstyle Use CheckStyle done

QQH0828 commented on src/main/java/store/Storage.java:
```diff
@@ -0,0 +1,112 @@
+        ArrayList<Task> arrayList = new ArrayList<>();
+        File file = new File(filePath);
+        Scanner scanner;
+        try {
+            scanner = new Scanner(file);
+            while (scanner.hasNext()) {
+                String nextLine = scanner.nextLine();
+                String[] taskLine = nextLine.trim().split(" \\| ");
+                switch (taskLine[0]) {
+                case "T":
+                    arrayList.add(ToDo.generateTask(taskLine));
+                    break;
+                case "E":
+                    arrayList.add(Event.generateTask(taskLine));
+                    break;
+                case "D":
+                    arrayList.add(DeadLine.generateTask(taskLine));
+                    break;
+                default:
+                    throw new DukeException("Invalid data!");
+                }
```
Ooh, I like your clear way of loading the date from the file!

seanfirefox replied:
Thanks!

QQH0828 commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,100 @@
+
+    public String getEnd() {
+        return end;
+    }
+    @Override
+    public String getTaskType() {
+        return "E";
+    }
```
Maybe can add one line before the @Override?

QQH0828 replied:
@Override, sorry for the typing.

QQH0828 reviewed (commented):
LGTM! Overall, I found your code easy to read with different packages and classes! And I personally feel that the codes follow the coding standard quite well!
Perhaps, the empty lines between each method can be the same, which will look more beautiful.

QQH0828 commented on src/main/java/command/AddTaskCommand.java:
```diff
@@ -0,0 +1,35 @@
+        super(inputArr);
+    }
+
+    @Override
+    public void execute(TaskList taskList, Ui ui, Storage storage) throws DukeException {
+
+    }
```
I notice that there is nothing inside the method, are you forgot to write it?

seanfirefox replied:
This is an abstract class so there is nothing meant to be in it, will leave a comment inside the method to specify. Thanks for noting it out!

QQH0828 commented on src/main/java/task/DeadLine.java:
```diff
@@ -0,0 +1,81 @@
+     * @return DeadLine Task.
+     * @throws DukeException Checks valid input from user.
+     */
+    public static DeadLine generate(String[] inputLine) throws DukeException {
+        if (inputLine.length < 2) {
+            throw new DukeException("Missing description");
+        }
+        String[] toPrintSplit = inputLine[1].split(" /by ", 2);
+        if (toPrintSplit.length < 2) {
+            throw new DukeException("Missing deadline");
+        }
+        return new DeadLine(toPrintSplit[0], false, DateTimeUtils.dateFormatter(toPrintSplit[1]));
```
I like your clever usage of splitting the input line to check the errors, it is easy to read and understands.

seanfirefox replied:
Thanks!

seanfirefox committed:
Level 10 GUI done

seanfirefox committed:
Merge branch 'branch-Level-10'

seanfirefox committed:
Exit GUI after bye user input

seanfirefox committed:
TaskList class: Add assert clause for postconditions in functions.

Checking assertions allows us to ensure the functions work as intended.

Add assert to document important assumptions that should hold at various points in the code.

Asserts are added as postconditions to check if the conditions work as intended.

seanfirefox committed:
Command classes: fix CLI output.

CLI output removed to be sent to GUI only previously.

Send output to both CLI and GUI so that code meant for CLI is not useless.

Make both CLI and GUI usable for Duke.

seanfirefox committed:
Merge pull request #2 from seanfirefox/branch-A-Assertions

A-Assertions done

seanfirefox committed:
Merge branch 'master' into branch-A-CodeQuality

seanfirefox committed:
Merge pull request #3 from seanfirefox/branch-A-CodeQuality

A-CodeQuality done

seanfirefox committed:
Fix merge conflict errors

seanfirefox committed:
C-FlexibleDataSource Flexible data source done

seanfirefox committed:
Merge pull request #4 from seanfirefox/branch-C-FlexibleDataSource

Add change file path command

seanfirefox committed:
Update README.md

seanfirefox committed:
Upload Ui.png

seanfirefox committed:
Set titlebar

seanfirefox committed:
Fix filepath handling

seanfirefox committed:
Change default filepath
